lib/src/shared/profiler.dart

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2020, the gRPC project authors. Please see the AUTHORS file
+// for details. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import '../client/channel.dart';
+
+/// Enable logging requests and response for clients.
```

Comment:
I think a much simpler API would be to have a single top-level variable in this file:

```
bool isTimelineLoggingEnabled = false;
```

then you don't need the rest of the code here.